src/providers/redux/userSlice.ts

```diff
@@ -0,0 +1,27 @@
+import { createSlice } from '@reduxjs/toolkit';
+import { IUserState } from '~/@types/redux';
+
+const initialState: IUserState = {
+  id: null,
+  name: '',
+  surname: '',
+  email: '',
+};
+
+const userSlice = createSlice({
+  name: 'user',
+  initialState,
+  reducers: {
+    updateUser: (state, action: { payload: IUserState }) => {
+      const { id, name, surname, email } = action.payload;
+      state.id = id;
+      state.name = name;
+      state.surname = surname;
+      state.email = email;
```

Comment:
Is there a better way to assign these values?

Reply:
Maybe :
`
state = {
id: id,
name: name,
surname: surname,
email: email,
}
`

Reply:
It does not work for some reason